src/components/MoviesCard/MoviesCard.js

```diff
@@ -1,39 +1,78 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import "./MoviesCard.css";
 import { useLocation } from "react-router-dom";
+import { MovieDuration } from "../../utils/constants";
 
-const MoviesCard = () => {
-  const [active, setActive] = useState(false);
-
+const MoviesCard = ({ movie, onMovieLike, onMovieDelete, savedMovies }) => {
+  const [active, setActive] = useState({});
   const location = useLocation();
   const isSavedMovies = location.pathname === "/saved-movies";
 
-  const toggleClick = () => {
-    setActive((prev) => !prev);
+  useEffect(() => {
+    if (!isSavedMovies) {
+      const isLikedCard = savedMovies.find((m) => m.movieId === movie.id);
+      if (isLikedCard) {
+        setActive(true);
```

Comment:
Initialize value is {} and passgin boolean? Btw you can write 
if(isLikedCard) setActive(!active)
pass default value false